### DISCUSSION TITLE
Prototype Archive page

### DISCUSSION THREAD

**Discussion by olbrich** [bridgetownrb/bridgetown#692] [General]
I haven't seen this addressed in the documentation, but is there a way to generate prototype pages for posts that groups them by publication date?  

I'd like to be able to generate a prototype page like '/archives/:year' and '/archives/:year/:month'.

In my case, the permalink for each post is '/:slug', with no date information.

**Comment by dcr8898** [comment]
This seems possible. I don't know if there is an official way, but I would do it with a custom taxonomy. The only issue I see is nesting months within years, as you suggested. If you really want that, I bet you could do it with a custom builder. I haven't done that yet, but here's how I would go about it with a taxonomy.

I would create a [custom taxonomy](https://www.bridgetownrb.com/docs/resources#taxonomies) called something like `year` or `date` or `date_slug` (whatever) . In each post, I would add this term to the front matter. Let's say I chose the term `year`, then the front matter would include `year: 2023`. If I want to group by year and month, then I would combine them with a different term, like `date: 2023-01`.

Finally, to generate the pages, I would create a [prototype page](https://www.bridgetownrb.com/docs/prototype-pages#simple-usage) and save it to `/src/archives/dates.html` (the filename doesn't matter, and the file format is up to you) with front matter something like this:

`````yaml
---
layout: default
title: Posts from :prototype-term
prototype:
  collection: posts
  term: year
`````
